### PR TITLE
refactor: add linting rules

### PR DESCRIPTION
**Pull request by dancrumb**
Following some comments from PR #111, I decided to create a smaller, more focused, PR around linting.

There are some good convention rules that linting can help bring to people's attentions

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    parserOptions: {
+        ecmaVersion: "latest",
+        sourceType: "module",
+        project: true,
+        tsconfigRootDir: __dirname
+    },
+    rules: {
+        "no-restricted-imports": ["error", "fs/promises"],
+        "func-style": ["error", "declaration", { "allowArrowFunctions": true }],
+        "max-params": ["error", 1]
```
Comment (dancrumb) [inline .eslintrc.js:19]: This was the best rule I could find to require parameter objects.

File: src/lib/resolveNodeModuleToDenoModule/resolveNodeModuleToDenoModule.ts [hunk 2]
```diff
@@ -41,16 +41,16 @@ type Result =
 
 export function resolveNodeModuleToDenoModuleFactory(
     params: {
-        userProvidedPorts: { [nodeModuleName: string]: string };
-        dependencies: { [nodeModuleName: string]: string };
-        devDependencies: { [nodeModuleName: string]: string };
+        userProvidedPorts: Dependencies;
+        dependencies: Dependencies;
+        devDependencies: Dependencies;
```
Comment (garronej) [inline src/lib/resolveNodeModuleToDenoModule/resolveNodeModuleToDenoModule.ts:46]: I'm not sure it's worth it to create a type for that its just a `Record<string, string>`.  
I've wrote it like `{ [nodeModuleName: string]: string }` to increace leagability.  
Having the type deported away makes the reader look elswhere...

File: src/lib/resolveNodeModuleToDenoModule/resolveNodeModuleToDenoModule.ts [hunk 3]
```diff
@@ -361,7 +363,7 @@ export const { getValidImportUrlFactory } = (() => {
 
         const { buildUrl } = buildUrlFactory({ moduleAddress });
 
-        const tsconfigJson = await fetch(buildUrl(gitTag, "tsconfig.json")).then(
+        const tsconfigJson = await fetch(buildUrl({ candidateBranchOrTag: gitTag, pathToFile: "tsconfig.json" })).then(
```
Comment (garronej) [inline src/lib/resolveNodeModuleToDenoModule/resolveNodeModuleToDenoModule.ts:366]:
```suggestion
        const tsconfigJson = await fetch(buildUrl({ "candidateBranchOrTag": gitTag, "pathToFile": "tsconfig.json" })).then(
```

**Review by garronej — COMMENTED** [review]
Dear @dancrumb,

First and foremost, I apologize for the delay in reviewing your pull request. Your understanding and patience are truly appreciated.

I recognize and appreciate the effort you've devoted to this task, and I'm in full agreement with you on the importance of standardizing our collaboration efforts. I can certainly see how incorporating eslint would be advantageous in achieving this objective.

That being said, given the current constraints, I regretfully have to admit that I don't have the capacity at present to execute a thorough configuration and align our CI/husky infrastructure accordingly.

Eslint, while useful, can indeed be demanding in terms of configuration to ensure its optimal effectiveness. Without the right adjustments, it may inadvertently hinder productivity and generate unnecessary noise, as we may be obliged to dismiss certain rules explicitly.

It is because of these considerations, and with the highest regard for the work you've put into this, that I find myself needing to decline this PR, despite my belief in the potential benefits of eslint.

I genuinely hope this decision does not discourage you from future contributions. Your insights and expertise are of great value, and I look forward to more collaboration with you.

Thank you once again for your understanding.

**Comment by dancrumb** [comment]
Hi Joseph,

Thanks for your consideration; please know that I know that you're doing
this in your own time, so I'll be as patient as needed.

It's a funny coincidence, actually. I was driving home (just before I got
this) and thinking that I should remove all the extraneous changes from my
other PR and just stick to the minimal relevant ones.

Thus, I'll update the other branch to pare it back and resubmit it when
that's done.

Thanks,

Dan

**Pull request closed without merging**